test/kino_db/connection_cell_test.exs

```diff
@@ -189,4 +191,36 @@ defmodule KinoDB.ConnectionCellTest do
     {:ok, conn} = Kino.start_child({MyXQL, opts})\
     """)
   end
+
+  test "password from secrets" do
+    {kino, _source} =
+      start_smart_cell!(ConnectionCell, %{
+        "variable" => "conn",
+        "type" => "postgres",
+        "port" => 5432
+      })
+
+    push_event(kino, "update_field", %{"field" => "password_secret", "value" => "true"})
+    assert_broadcast_event(kino, "update", %{"fields" => %{"password_secret" => "true"}})
+
+    push_event(kino, "update_field", %{"field" => "password_from_secret", "value" => "LB_PASS"})
+    assert_broadcast_event(kino, "update", %{"fields" => %{"password_from_secret" => "LB_PASS"}})
```

Comment:
We could merge this into `"password_secret" => string | nil`, especially that we need the same for any other secret, wdyt?

Reply:
Hey @jonatanklosko! My bad! I saw your comment only today. I don't think I understood what you meant 🤔

Reply:
I mean that instead of having two fields (a boolean and a value), we could have just a single field that is either nil or a string (so we don't need the additional boolean). This way we reduce the amount of fields, especially as we need to support more secrets.

Reply:
We have a checkbox to toggle between input and secrets. We're using the boolean to keep the UI of all clients synchronized independently of the values. We also want to keep both values during the session, so resetting the secret to nil when deselecting the checkbox is not an option. Also, Valim has some plans for dealing with empty or unavailable secrets. I don't think it's possible to achieve all this with a single field 🤔 wdyt?

Reply:
I see, if we need both values this makes sense. So then I would suggest that we have the following form fields:

* `password` - plain text
* `password_secret` - name of the secret to use
* `use_password_secret` - boolean

Then, when converting to attributes we keep either `password` or `password_secret`. We don't need to persist `use_password_secret` at all, because on init we can do `"use_password_secret" => Map.has_key?(attrs, "password_secret)`.

Does this make more sense?

Reply:
Yep!